### PR TITLE
GTH: Fix QPLL module instantiation & introduce freq(TXUSRCLK)=freq(TXOUTCLK)/2

### PR DESCRIPTION
**Pull request by HarryMakes**
## Summary

* Reinforce the distinction between PLL representations (`ChannelPPL`, `QuadPLL`) and Xilinx building blocks (`GTHE3_CHANNEL`, `GTHE3_COMMON`)
* Emphasise on difference in configuration of Xilinx objects between CPLLs and QPLLs
* Introduce frequency divider on GTH output clocks for 80-bit data width
* Fix parameters previously overlooked for using Xilinx objects

## Background

The main rationale behind this PR is to adopt a higher sampling frequency (1 GS/s) for JESD204B DACs in Sayma running at 125 MHz RTIO and JESD clock (see https://github.com/HarryMakes/artiq/commit/sayma1g). The line rate needs to be raised from 6 Gb/s, which is compatible with GTH's CPLL, to 10 Gb/s, which is compatible with GTH's QPLL0. Since data arrives at the JESD core at 125 MHz on the transport layer, a total of 10e9/125e6 = 80 bits of data need to be transmitted from the transceiver per JESD clock cycle, doubling the port width needed (which is 40 bits currently). Under the 8b/10b encoding, this is equivalent to having 8 bytes of data arriving at the transceiver per JESD clock cycle.

Therefore, the first task to do is to properly instantiate Xilinx libraries for QPLLs to work properly on GTH. The current implementation of QPLLs seem to guide the user to instantiate one GTHE3_COMMON representing one quad for each QPLL channel. This would be incorrect because a quad should be connected to a maximum of 4 PLL channels in the same clock region, which also boils down to the difference in configuration between CPLLs and QPLLs. This led to my proposal of separating the instantiation functions for GTHE3_COMMON and GTHE3_CHANNEL.

The second task is to adjust the clocking with respect to GTH. GTH imposes that under 8-byte mode, TXUSRCLK and TXUSRCLK2 must be driven by TXOUTCLK with a ÷1 divider and ÷2 divider respectively. Therefore, an additional BUFG_GT buffer would be needed to divide TXOUTCLK to a new clock domain (`tx_half`) for each transceiver. This is the reason I chose to introduce the option to switch to a divided clock when instantiating a GTH transceiver object.

### CODE REVIEW

**Comment by sbourdeauducq** [comment]
LGTM. Did you test that there are no regressions/backward compatibility issues with the legacy 600Msps/2.4GHz code (in ARTIQ release-5)?

**Comment by HarryMakes** [comment]
Backward compatibility with `master`'s 600 MS/s has been tested, using changes listed on https://github.com/m-labs/artiq/pull/1568.

File: jesd204b/phy/__init__.py [hunk 1]
```diff
@@ -1,4 +1,5 @@
 from migen import *
+# from migen.genlib.cdc import Gearbox
```
Comment (sbourdeauducq) [inline jesd204b/phy/__init__.py]: remove commented out code

Comment (HarryMakes) [reply jesd204b/phy/__init__.py]: Fixed with d7a67aae9bbcd5058fae706770562d11da25932e.

**Pull request merged**